futures-macro-async/src/lib.rs

```diff
@@ -191,7 +191,15 @@ if_nightly! {
         });
         syn::token::Semi([block.brace_token.0]).to_tokens(&mut result);
 
+        let await = if pinned {
+            await()
+        } else {
+            await_move()
+        };
+
         let gen_body_inner = quote_cs! {
+            #await
```

Comment:
What happens here if you declare an `#[async]` function inside an `#[async_move]` function (or vice versa)?

Reply:
Good question, I assumed macros would follow normal scoping/shadowing rules and this would work. Macros have enough _unique_ behaviour that I'm not certain though, I'll add a test covering this to check.

Reply:
See the test [`qux2`](https://github.com/rust-lang-nursery/futures-rs/pull/925/files#diff-27836183fc84a021f648e6405d935276R31). I believe this should error if the macro were to leak either from inner -> outer function or outer -> inner function.